core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -94,12 +91,14 @@ private[spark] abstract class YarnSchedulerBackend(
   /**
    * An actor that communicates with the ApplicationMaster.
    */
-  private class YarnSchedulerActor extends Actor {
+  protected class YarnSchedulerActor(isDriver: Boolean)  extends Actor {
     private var amActor: Option[ActorRef] = None
 
     override def preStart(): Unit = {
       // Listen for disassociation events
-      context.system.eventStream.subscribe(self, classOf[RemotingLifecycleEvent])
+      if (!isDriver) {
+        context.system.eventStream.subscribe(self, classOf[RemotingLifecycleEvent])
+      }
```

Comment:
because in cluster mode, YarnSchedulerActor and driver  use same actorSystem, if YarnSchedulerActor subscribe and listen driver's actor, messages from driver's actor that should to send to executor will send to YarnSchedulerActor .so there is a big wrong and we cannot listen for cluster mode.